modules/classify_novel_taxa.nf

```diff
@@ -0,0 +1,429 @@
+// This workflow is for trying to find and classify possibly novel RNA viruses that might have been overlooked by kraken. 
+// 1) runs a metagenomic assembly on short or long reads,
+// 2) runs VirBot or GeNomad classification on them.		     
+// Output: file with contigs identified as RNA viral 'viral_contigs.fa',
+//         contig id to taxonomy mapping file 'tax_assignments.tsv',
+//         (optional) {assembler}_assembly_stats.txt file with the performed assembly statistics - 
+//         all in the 'discovery' directory
+
+// Related options and params:
+// --paired) default: false
+// --read_type) 'illumina'|'ont'
+// --assembler) for short (Illumina) reads: 'megahit' (default), 'rnaspades'
+//              for long (Oxford Nanopore) reads: 'rnabloom' (default), 'flye'
+// Notes: megahit is faster while rnaspades tends to produce more contiguous assembly 
+//	  megahit here allows short contigs (>=150 bp) whereas rnaspades does not,
+//		so in the case of low-quality sequincing data the assembly with
+//		rnaspades might lead to lower recall
+//        rnaspades results in the identification of more sequences of retroviral origin
+//        flye assembly requires high coverage with reads (at least ~x30) to complete, 
+//		so it was common to fail, and FAILURE IS NOT HANDLED.
+//	  rnabloom will fail if there are very few reads, again, FAILURE IS NOT HANDLED.
+//		Also it might produce an empty fasta file if all the resulting contigs have
+//		length below the threshold of 200 bp. This will fail with genomad but not with virbot 
+//        all mentioned assemblers (rnabloom, flye, megahit, spades) should work on gzipped files
+// --classifier) 'virbot'|'genomad', default: virbot
+// Notes: genomad requires longer contigs for its neural network to run,
+//		so the contigs are filtered by length (>=2kbp) beforehand.
+//		Therefore, it is adviced to use genomad on long read data or relatively contiguous assembly
+//        overall, at times virbot has greater sensitivity while genomad might be more precise (?), 
+//		but generally they produce similar output
+//        virbot does classification to the lowest level possible
+//        genomad here is run on preset 'relaxed' (no end filtering)
+//        for genomad, the assignments other than 'Riboviria' (including those labeled as 'Unclassified') 
```

Comment:
You say assignments other than Riboviria are excluded. Is this a default of genomad? Is it designed just for Riboviria?

Reply:
It is not default, I filter out the other assignments in the resulting file to be concurrent with virbot and the module's purpose